client/src/components/ai-enhanced/EnhancedAIAssistant.tsx

```
@@ -235,15 +236,18 @@ export const EnhancedAIAssistant: React.FC<EnhancedAIAssistantProps> = ({
         case 'token':
           setStatus('streaming');
           setIsTyping(true);
-          setStreamBuf((s) => s + evt.value);
+          const nextBuf = streamBufRef.current + evt.value;
+          streamBufRef.current = nextBuf;
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)